python/sidecar/ingest.py

```diff
@@ -26,90 +32,210 @@
 
-    def run(self):
+        self.references = self._load_references()
+
+    def run(self) -> None:
         logger.info(f"Starting ingestion for project: {self.project_name}")
-        logger.info(f"Input directory: {self.input_dir}")
-        self._create_directories()
-        self.call_grobid_server()
-        self.convert_grobid_xml_to_json()
+
+        self._copy_uploads_to_staging()
+        self._call_grobid_for_staging()
+        self._convert_grobid_xml_to_json()
+
         references = self.create_references()
         self.save_references(references)
+
         response = self.create_response_from_references(references)
         sys.stdout.write(response.json())
+
+        for ref in references:
+            self._remove_temporary_files_for_reference(ref)
+
         logger.info(f"Finished ingestion for project: {self.project_name}")
-        logger.info(f"Response: {response}")
 
     def _create_directories(self) -> None:
+        if not self.staging_dir.exists():
+            self.staging_dir.mkdir()
+
         if not self.grobid_output_dir.exists():
             self.grobid_output_dir.mkdir()
 
         if not self.storage_dir.exists():
             self.storage_dir.mkdir()
+    
+    def _check_for_uploaded_files(self) -> bool:
+        """
+        Checks for PDF files in the `uploads` directory.
+
+        Returns
+        -------
+        bool
+            True if PDF files are in the `uploads` directory.
+        """
+        if not self.uploaded_files:
+            logger.warning(f"No PDF files found in directory: {self.input_dir}")
+            return False
+        return True
+
+    def _check_for_staging_files(self) -> bool:
+        """
+        Checks for PDF files in the `.staging` directory.
+
+        Returns
+        -------
+        bool
+            True if PDF files are in the `.staging` directory.
+        """
+        if not list(self.staging_dir.glob("*.pdf")):
+            logger.warning(f"No PDF files found in directory: {self.staging_dir}")
+            return False
+        return True
+
+    def _load_references(self) -> list[Reference]:
+        """
+        Loads existing References list for project. If none exist, this will 
+        be an empty list.
+
+        References that have been previously processed will not be ingested 
+        again.
+        """
+        filepath = self.storage_dir.joinpath('references.json')
+        if not filepath.exists():
+            logger.warning(f"No previous `references.json` found at {filepath}")
+            return []
+
+        jstore = JsonStorage(filepath)
+        jstore.load()
+        return jstore.references
 
-    def get_files_to_ingest(self) -> list[Path]:
+    def _get_files_to_ingest(self) -> list[Path]:
         """
-        Determines which files need to be ingested
-        :return: bool
+        Determines which files need to be ingested by comparing the list of 
+        already processed References against files found in `uploads`.
```

Comment:
Wouldn't be easier to give you the list of file paths to ingest? In the FE the user selects files and not directories.